### ISSUE TITLE
Crash with UITextView

### ISSUE THREAD

**Issue by mshibanami** [zwaldowski/BlocksKit#257]
1. Open ![this zip file](https://github.com/mshibanami/mshibanami/raw/9757df1a304d64dc2328523bfbe65385d788874a/Sample.zip) and execute
   `pod install`
2. Open `Sample.xcworkspace` and run it
3. Select "あああ" and show menu
4. Tap "Learn..." Button

![skitched-20140830-110340](https://cloud.githubusercontent.com/assets/1333214/4097875/e0357f98-2fe9-11e4-9192-33cf457d2f42.jpg)

Then App crashed with the following error.

```
-[TIUserDictionaryController loadPersistentStoreAtURL:] could not open user dictionary at file:///var/mobile/Library/Keyboard/CoreDataUbiquitySupport/mobile~***/UserDictionary/***/store/CloudUserDictionary.sqlite: Error Domain=NSCocoaErrorDomain Code=256 "The operation couldn’t be completed. (Cocoa error 256.)" UserInfo=0x14dd09f0 {reason=Failed to access file: 1}
```

![skitched-20140830-104707](https://cloud.githubusercontent.com/assets/1333214/4097878/17c231ea-2fea-11e4-9e8e-1863bf763de9.jpg)

**Comment by muhuaxin** [comment]
I get the same problem. 

if (realDelegate && [realDelegate respondsToSelector:@selector(textFieldShouldBeginEditing:)])
		ret = [realDelegate textFieldShouldBeginEditing:textField];

It is a recursive call. 

Is it solved?

**Comment by yanzhiwei147** [comment]
Yes,I get the same problem. I found a recursive call at `UITextField+BlocksKit`,What‘s the time can solve it?

**Comment by aipeople** [comment]
I found and solved another recursive call problem (#316). Do you guys think it is the same problem ?

**Comment by atomicink** [comment]
I got this bug too.

**Comment by alexlee002** [comment]
@zwaldowski:  why if user did defined `bk_XxxBlock`(e.g.: bk_shouldClearBlock), we still call the original delegate's method? 

I think this is the root of the crash.  In my opinion, if user defies `bk_xxxBlock`, we just call this block and DO NOT call the original delegate method again.

**Comment by ppaulojr** [comment]
This is happening if you use BlocksKit and [PPJEmailPicker](https://github.com/ppaulojr/PPJEmailPicker) in the same project.

It crashes after entering an infinity loop in `forwardInvocation` 

![captura de tela 2016-05-26 as 16 11 40](https://cloud.githubusercontent.com/assets/1206478/15586793/9cc5da88-235c-11e6-81f2-2fa42cd471f4.png)

Any thoughts?

**Comment by MoLice** [comment]
This also happening if you use BlocksKit and [QMUITextField](https://github.com/QMUI/QMUI_iOS/blob/master/QMUIKit/UIKitExtensions/QMUITextField.h) in the same project, just like @ppaulojr said.
